### PR TITLE
Update cloud testing for Elastic Agent instructions

### PR DESCRIPTION
**Pull request by simitt**
## Motivation/summary

Make it easier for using the cloud testing infrastructure for a deploying a custom Elastic Agent image. 

## Checklist

- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

### CODE REVIEW

File: testing/cloud/README.md [hunk 1]
```diff
@@ -7,9 +7,32 @@ off the SNAPSHOT images referenced in docker-compose.yml.
...
 updating an existing deployment.
+
+After finishing the testing, run `make destroy` for deleting the clusters. 
+
+## Building and deploying a custom Elastic Agent image
+
+For testing a non-released Elastic Agent image, the cloud base image needs to be configured via `ELASTIC_AGENT_DOCKER_IMAGE` and `ELASTIC_AGENT_IMAGE_TAG`. 
+```
+ELASTIC_AGENT_DOCKER_IMAGE=docker.elastic.co/observability-ci/elastic-agent-cloud ELASTIC_AGENT_IMAGE_TAG=a8b36d05919c40385fe5a28d30225a112626d48b make elastic_agent_docker_image docker_image.auto.tfvars
+ECE_API_KEY=<your-api-key> make apply
```
Comment (kruskall) [inline testing/cloud/README.md:25]: **question**: did you mean `EC_API_KEY` ?

File: testing/cloud/README.md [hunk 2]
```diff
@@ -7,9 +7,32 @@ off the SNAPSHOT images referenced in docker-compose.yml.
...
+NOTE: Before running the above command, ensure that the file `docker_image.auto.tfvars` does not exist, as it won't be overwritten.
+
+Take a look at the `variables.tf` for specifying sizes, version, etc. 
+
+Example - testing local changes for Elastic Agent:
+First, build the elastic agent docker image from your local branch and push it to the internal docker registry, e.g. `docker.elastic.co/observability-ci/elastic-agent:8.6.1-8f6887b9-SNAPSHOT-simitt`.
+In this case, the command to build the docker image and create the deployment would look like this: 
+```
+ELASTIC_AGENT_DOCKER_IMAGE=docker.elastic.co/observability-ci/elastic-agent-cloud ELASTIC_AGENT_IMAGE_TAG=a8b36d05919c40385fe5a28d30225a112626d48b make elastic_agent_docker_image docker_image.auto.tfvars
+ECE_API_KEY=<your-api-key> make apply
```
Comment (kruskall) [inline testing/cloud/README.md:36]: Same as above

**Review by kruskall — COMMENTED** [review]
Small question about `ECE_API_KEY`

**Review by kruskall — APPROVED** [review]
LGTM

**Review by marclop — APPROVED** [review]

**Pull request merged**